psi4/src/psi4/cc/ccenergy/BT2.cc

```diff
@@ -121,11 +119,11 @@ void CCEnergyWavefunction::BT2() {
             global_dpd_->buf4_mat_irrep_init(&tau, 0);
             global_dpd_->buf4_mat_irrep_rd(&tau, 0);
             tau_diag = global_dpd_->dpd_block_matrix(tau.params->rowtot[0], moinfo_.nvirt);
-            for (ij = 0; ij < tau.params->rowtot[0]; ij++)
-                for (Gc = 0; Gc < moinfo_.nirreps; Gc++)
-                    for (C = 0; C < moinfo_.virtpi[Gc]; C++) {
-                        c = C + moinfo_.vir_off[Gc];
-                        cc = tau.params->colidx[c][c];
+            for (int ij = 0; ij < tau.params->rowtot[0]; ij++)
+                for (int Gc = 0; Gc < moinfo_.nirreps; Gc++)
+                    for (int C = 0; C < moinfo_.virtpi[Gc]; C++) {
+                        auto c = C + moinfo_.vir_off[Gc];
+                        auto cc = tau.params->colidx[c][c];
```

Comment:
@loriab I think that will depend mostly on the type of the variable, if copy/move assign are defined and how they are implemented.  

Someone else can correct me if I am wrong, but in this particular case, declaration + assign inside is going to be better if there is any difference at all.

Reply:
Because I was curious the generated assembly for a simple example is identical with different 
 offsets for stack allocated vars with `-O0` and with anything higher stuff starts to get optimized away, but the generated assembly is still identical with different offsets.  https://godbolt.org/z/63O-yl

Reply:
thanks for that link, btw. I played with it two weeks ago. will try to readjust my instincts.